src/Microsoft.AspNet.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.AspNet.Server.Kestrel.Infrastructure;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http
+{
+    /// <summary>
+    /// Manages the generation of the date header value.
+    /// </summary>
+    public class DateHeaderValueManager : IDisposable
+    {
+        private readonly ISystemClock _systemClock;
+        private readonly TimeSpan _timeWithoutRequestsUntilIdle;
+        private readonly TimeSpan _timerInterval;
+        private readonly uint _timerTicksWithoutRequestsUntilIdle;
```

Comment:
`double`? and _timerTicksSinceLastRequest can be `double` as well to avoid casting issues & equality check between `int` and `uint`.  And equality check can become comparison to avoid double precision issues.